examples/lock-app/nrfconnect/README.md

```diff
@@ -1,293 +1,336 @@
 
 <hr>
 
--   [CHIP nRF52840 Lock Example Application](#chip-nrf52840-lock-example-application)
-    -   [Introduction](#introduction)
-    -   [Device UI](#device-ui)
-    -   [Building](#building)
-        -   [Using Docker container](#using-docker-container)
-        -   [Using Native shell](#using-native-shell)
-        -   [Supported nRF Connect SDK versions](#supported-nrf-connect-sdk-versions)
-    -   [Configuring the example](#configuring-the-example)
-    -   [Flashing and debugging](#flashing-and-debugging)
-    -   [Accessing the command line](#accessing-the-command-line)
+-   [Overview](#overview)
+    -   [Bluetooth LE Advertising](#bluetooth-le-advertising)
+    -   [Bluetooth LE Rendezvous](#bluetooth-le-rendezvous)
+    -   [Thread provisioning](#thread-provisioning)
+-   [Requirements](#requirements)
+-   [Device UI](#device-ui)
+-   [Setting up environment](#setting-up-environment)
+    -   [Using Docker container](#using-docker-container)
+    -   [Using Native shell](#using-native-shell)
+-   [Building](#building)
+-   [Flashing and debugging](#flashing-and-debugging)
+-   [Testing the example](#testing-the-example)
 
 <hr>
 
-<a name="intro"></a>
+<a name="overview"></a>
+
+## Overview
+
+This example is running on the nRF Connect platform, which is based
+on the [nRF Connect SDK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/index.html)
+and [Zephyr RTOS](https://zephyrproject.org/). Visit CHIP's
+[nRF Connect Platform Overview](TODO:...) to read more information about
+platform structure and dependencies.
+
+The CHIP device that runs the lock application is controlled by the CHIP controller
+device over the Thread protocol. By default, the CHIP device should have Thread disabled,
+and it should be paired with CHIP controller and get configuration from it.
+Some actions required before establishing full communication are described below.
+
+The example also comes with a test mode, which allows to start Thread with the default settings
+by pressing button manually. However, this mode does not guarantee that the device will be able
+to communicate with the CHIP controller and other devices.
+
+### Bluetooth LE Advertising
+
+After powering up the device for the first time, it will start advertising over
+Bluetooth LE to inform other devices about its presence. For security reasons,
+Bluetooth LE advertising won't start automatically after powering up the device.
+To make the device discoverable, you must press **Button 4**.
+
+### Bluetooth LE Rendezvous
+
+In CHIP, there is a commissioning procedure called Rendezvous, which is done over
+Bluetooth LE between a CHIP device and the CHIP controller,
+where the controller has the commissioner role.
+
+To start the Rendezvous, the controller must get the onboarding information from the CHIP device.
+The data payload is encoded within a QR code and typically presented on the device's display.
+For this example, however, it is shared using **NFC**.
```

Comment:
My comment somehow was lost.. but, this sentence is not true. For this example you can use either QR Code scanning with pressing Button 4, or NFC - both can work without changing the firmware.

Reply:
Actually it's quite weird, as I'm almost 100% sure that I pushed commit fixing it yesterday. I checked now and it was in my git log, but after git pull it disappeared. @greg-fer didn't you squashed and deleted it accidentialy?

Reply:
I might've, sorry for this. Had some big issues with merge conflicts, and how the conflict was shown made it impossible to find the edits (two big chunks of HEAD/commit covering the whole page).

Reply:
Anyway, if we have two options: QR Code AND Button 4 or NFC, then it is not clear in the docs at all.